manager.py

```diff
@@ -66,31 +66,31 @@ def configure(self, context):
             self.get_dependency('ConfigurationManager')
 
         # fetch component settings
-        self.config_api_url_prefix = \
+        self._config_api_url_prefix = \
             Settings.get("configuration", "config_api_url_prefix",
-                fallback="https://api.n.io/v1/instance_configurations")
-
-        setting_config_id = Settings.get("configuration", "config_id",
-            fallback=None)
-        self._config_id = Persistence().load("configuration_id",
-                                            default=setting_config_id)
-        setting_config_version_id = \
-            Settings.get("configuration", "config_version_id",
-                fallback=None)
-        self._config_version_id = Persistence().\
-            load("configuration_version_id", default=setting_config_version_id)
-        
+                         fallback="https://api.n.io/v1/instance_configurations")
+
+        self._config_id = Persistence().load(
+            "configuration_id",
+            default=Settings.get("configuration", "config_id"))
+        self._config_version_id = Persistence().load(
+            "configuration_version_id",
+            default=Settings.get("configuration", "config_version_id"))
+
         self._start_stop_services = Settings.getboolean(
             "configuration", "start_stop_services", fallback=True)
         self._delete_missing = Settings.getboolean(
             "configuration", "delete_missing", fallback=False)
 
         # fetch instance specific settings
-        default = Persistence().load("api_key", default=None)
-        self.api_key = NIOEnvironment.get_variable('API_KEY', 
-                                                   default=default)
-
-        # config autonomy spcific settings
+        self._api_key = Persistence().load(
+            "api_key",
+            default=Settings.get("configuration", "instance_api_key"))
```

Comment:
We would still want to have a way of pulling `API_KEY` from the NIOEnvironment for cloud instances